From ad157b7e5a3341a240eeaab71705501682e8cd05 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Mon, 27 Mar 2017 15:42:58 +0800
Subject: [PATCH 1/2] ASoC: fsl_hdmi: update fsl_hdmi driver to adapt to new
 hdmi codec

Since commit 9529138276c85 ("ASoC: hdmi: Remove obsolete dummy HDMI codec") delete
old hdmi codec source code and commit 09184118a8ab ("ASoC: hdmi-codec: Add hdmi-codec
for external HDMI-encoders") introduce new hdmi codec source code, we need to update
fsl_hdmi driver to adapt to new hdmi codec.

Signed-off-by: Wang Quanyang <quanyang.wang@windriver.com>
---
 sound/soc/fsl/fsl_hdmi.c |   27 +++++++++++++++++++++++++--
 sound/soc/fsl/imx-hdmi.c |    4 ++--
 2 files changed, 27 insertions(+), 4 deletions(-)

diff --git a/sound/soc/fsl/fsl_hdmi.c b/sound/soc/fsl/fsl_hdmi.c
index 4f2ac27..2798ff3 100644
--- a/sound/soc/fsl/fsl_hdmi.c
+++ b/sound/soc/fsl/fsl_hdmi.c
@@ -34,6 +34,7 @@
 #include <sound/pcm.h>
 #include <sound/soc.h>
 #include <sound/asoundef.h>
+#include <sound/hdmi-codec.h>
 
 #include <video/mxc_hdmi.h>
 
@@ -600,6 +601,28 @@ static const struct snd_soc_component_driver fsl_hdmi_component = {
 	.name		= "fsl-hdmi",
 };
 
+static int imx_hdmi_audio_hw_params(struct device *dev, void *data,
+				    struct hdmi_codec_daifmt *daifmt,
+				    struct hdmi_codec_params *params)
+{
+	return 0;
+}
+
+static void imx_hdmi_audio_shutdown(struct device *dev, void *data)
+{
+}
+
+static const struct hdmi_codec_ops imx_hdmi_audio_codec_ops = {
+	.hw_params = imx_hdmi_audio_hw_params,
+	.audio_shutdown = imx_hdmi_audio_shutdown,
+};
+
+static struct hdmi_codec_pdata codec_data = {
+	.ops = &imx_hdmi_audio_codec_ops,
+	.max_i2s_channels = 8,
+	.i2s = 1,
+};
+
 static int fsl_hdmi_dai_probe(struct platform_device *pdev)
 {
 	struct device_node *np = pdev->dev.of_node;
@@ -654,8 +677,8 @@ static int fsl_hdmi_dai_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	hdmi_data->codec_dev = platform_device_register_simple(
-			"hdmi-audio-codec", -1, NULL, 0);
+	hdmi_data->codec_dev = platform_device_register_data(NULL,
+			"hdmi-audio-codec", PLATFORM_DEVID_AUTO, &codec_data, sizeof(codec_data));
 	if (IS_ERR(hdmi_data->codec_dev)) {
 		dev_err(&pdev->dev, "failed to register HDMI audio codec\n");
 		ret = PTR_ERR(hdmi_data->codec_dev);
diff --git a/sound/soc/fsl/imx-hdmi.c b/sound/soc/fsl/imx-hdmi.c
index a66ea0d..e7e65fc 100644
--- a/sound/soc/fsl/imx-hdmi.c
+++ b/sound/soc/fsl/imx-hdmi.c
@@ -26,8 +26,8 @@
 static struct snd_soc_dai_link imx_hdmi_dai_link = {
 	.name = "i.MX HDMI Audio Tx",
 	.stream_name = "i.MX HDMI Audio Tx",
-	.codec_dai_name = "hdmi-hifi",
-	.codec_name = "hdmi-audio-codec",
+	.codec_dai_name = "i2s-hifi",
+	.codec_name = "hdmi-audio-codec.0.auto",
 	.platform_name = "imx-hdmi-audio",
 };
 
-- 
1.7.5.4

